docs/federate.md

```diff
@@ -1,163 +1,41 @@
-Setting up Federation
+Setting up federation
 =====================
 
 Federation is the process by which users on different servers can participate
 in the same room. For this to work, those other servers must be able to contact
 yours to send messages.
 
-The ``server_name`` configured in the Synapse configuration file (often
-``homeserver.yaml``) defines how resources (users, rooms, etc.) will be
-identified (eg: ``@user:example.com``, ``#room:example.com``). By
-default, it is also the domain that other servers will use to
-try to reach your server (via port 8448). This is easy to set
-up and will work provided you set the ``server_name`` to match your
-machine's public DNS hostname, and provide Synapse with a TLS certificate
-which is valid for your ``server_name``.
+The `server_name` configured in the Synapse configuration file (often
+`homeserver.yaml`) defines how resources (users, rooms, etc.) will be
+identified (eg: `@user:example.com`, `#room:example.com`). By default,
+it is also the domain that other servers will use to try to reach your
+server (via port 8448). This is easy to set up and will work provided
+you set the `server_name` to match your machine's public DNS hostname.
+
+You will also need a valid TLS certificate for this `server_name` served
+on port 8448 - the preferred way to do that is by using a reverse proxy,
+see [reverse_proxy.md](<reverse_proxy.md>) for instructions on how to
+correctly set one up.
+
+In some cases you might not want Synapse to be running on the machine that
+has the `server_name` as its public DNS hostname, or federation traffic
+to use port than 8448 (e.g. you want to use `example.com` as your `server_name`
+but want Synapse to be reachable on `synapse.example.com:443`). This can
+be done using delegation, which allows an admin to dictate where federation
+traffic should be sent, see [delegate.md](<delegate.md>) for instructions on
+how to set this up.
```

Comment:
> In some cases you might not want to run Synapse on the machine that
has the `server_name` as its public DNS hostname, or you might want federation traffic
to use a different port. For example, you might want to have your user names look like `@user:example.com`, but you want to run Synapse on `synapse.example.com` on port 443. This can
be done using delegation, which allows an admin to control where federation
traffic should be sent. See [delegate.md](<delegate.md>) for instructions on
how to set this up.